### ISSUE TITLE
Cannot remove node due to message size too large

### ISSUE THREAD

**Issue by IgorOhrimenko** [moby/moby#45027]
### Description

```
docker node ls
ID                            HOSTNAME       STATUS    AVAILABILITY   MANAGER STATUS   ENGINE VERSION
t4tuaimodnterp655mcmvep5s     manager-00     Ready     Pause          Reachable        20.10.23
9a7k4zuct2uo6ly62dfvdat0d     manager-01     Ready     Pause          Reachable        23.0.1
7wzpf1et0x8bsuc4788p3loxe *   manager-02     Ready     Pause          Leader           20.10.21
k66y0ue46m4s2izr5ixx02pho     node-01        Down      Drain                           20.10.17
nie23fzaxyxbxagunkst5d4wx     node-02        Ready     Active                          20.10.23
xyjvw1u15ir6mniks70vb4e13     node-03        Ready     Active                          20.10.14
5rcg19yl5xjstm90fnlsx2m0k     node-04        Ready     Active                          20.10.14
2szicdrvsa227vuwoq82rjnz7     node-05        Ready     Active                          20.10.14
s4yysgfr7x10fut0lsn9oukb0     node-06        Down      Pause                           23.0.0
9hom5xlhu5kf6ha2emokt8t2j     node-07        Ready     Active                          20.10.17
yrdij8xzlecm3bv68gnfp9enl     node-08        Ready     Active                          20.10.22
5nee3fbzfvbre11rn74p2nnjs     node-09        Ready     Active                          20.10.22
```

First, I tried remove node-01: `docker node rm node-01` and get error Error response from daemon: rpc error: code = Unknown desc = raft: raft message is too large and can't be sent

```
Feb 17 03:42:03 manager-02 level=error msg="Error removing node node-01: rpc error: code = Unknown desc = raft: raft message is too large and can't be sent"
Feb 17 03:42:03 manager-02 level=error msg="Handler for DELETE /v1.41/nodes/node-01 returned error: rpc error: code = Unknown desc = raft: raft message is too large and can't be sent"
```
Second, I tried `docker swarm ca --rotate` and it hung on the stage `rotated TLS certificates`:
```
desired root digest: sha256:a183f59771cadfa5a423905c363ce97ef1b8cee269a4dde3940067bcd37c31a1
  rotated TLS certificates:  [=========================================>         ] 10/12 nodes
  rotated CA certificates:   [>                                                  ] 0/12 nodes
```
Below there are logs from all `Active` nodes:

<details>
    <summary>syslog dockerd logs</summary>

```
Feb 17 03:42:20 manager-00 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 manager-00 level=info msg="forced certificate renewal" module=node/tls node.id=t4tuaimodnterp655mcmvep5s node.role=swarm-manager
Feb 17 03:42:21 manager-00 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:22 manager-00 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:22 manager-00 level=info msg="agent: found node update" module=node/agent node.id=t4tuaimodnterp655mcmvep5s
Feb 17 03:42:22 manager-00 level=info msg="manager selected by agent for new session: { }" module=node/agent node.id=t4tuaimodnterp655mcmvep5s
Feb 17 03:42:22 manager-00 level=info msg="waiting 0s before registering session" module=node/agent node.id=t4tuaimodnterp655mcmvep5s
Feb 17 03:42:22 manager-00 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 manager-01 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 manager-01 level=info msg="forced certificate renewal" module=node/tls node.id=9a7k4zuct2uo6ly62dfvdat0d node.role=swarm-manager
Feb 17 03:42:21 manager-01 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 manager-01 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 manager-01 level=info msg="agent: found node update" module=node/agent node.id=9a7k4zuct2uo6ly62dfvdat0d
Feb 17 03:42:21 manager-01 level=info msg="[core] [Channel #15] Channel Connectivity change to SHUTDOWN" module=grpc
Feb 17 03:42:21 manager-01 level=info msg="[core] [Channel #15 SubChannel #16] Subchannel Connectivity change to SHUTDOWN" module=grpc
Feb 17 03:42:21 manager-01 level=info msg="[core] [Channel #15 SubChannel #16] Subchannel deleted" module=grpc
Feb 17 03:42:21 manager-01 level=info msg="[core] [Channel #15] Channel deleted" module=grpc
Feb 17 03:42:21 manager-01 level=info msg="manager selected by agent for new session: { }" module=node/agent node.id=9a7k4zuct2uo6ly62dfvdat0d
Feb 17 03:42:21 manager-01 level=info msg="waiting 0s before registering session" module=node/agent node.id=9a7k4zuct2uo6ly62dfvdat0d
Feb 17 03:42:22 manager-01 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 manager-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 manager-02 level=info msg="forced certificate renewal" module=node/tls node.id=7wzpf1et0x8bsuc4788p3loxe node.role=swarm-manager
Feb 17 03:42:21 manager-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:22 manager-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:22 manager-02 level=info msg="agent: found node update" module=node/agent node.id=7wzpf1et0x8bsuc4788p3loxe
Feb 17 03:42:22 manager-02 level=info msg="manager selected by agent for new session: { }" module=node/agent node.id=7wzpf1et0x8bsuc4788p3loxe
Feb 17 03:42:22 manager-02 level=info msg="waiting 0s before registering session" module=node/agent node.id=7wzpf1et0x8bsuc4788p3loxe
Feb 17 03:42:23 manager-02 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-02 level=info msg="forced certificate renewal" module=node/tls node.id=nie23fzaxyxbxagunkst5d4wx node.role=swarm-worker
Feb 17 03:42:21 worker-02 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.8:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-02 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-02 level=info msg="agent: found node update" module=node/agent node.id=nie23fzaxyxbxagunkst5d4wx
Feb 17 03:42:21 worker-02 level=error msg="agent: session failed" backoff=100ms error="rpc error: code = Canceled desc = context canceled" module=node/agent node.id=nie23fzaxyxbxagunkst5d4wx
Feb 17 03:42:21 worker-02 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-02 level=info msg="manager selected by agent for new session: {9a7k4zuct2uo6ly62dfvdat0d 10.10.10.7:2377}" module=node/agent node.id=nie23fzaxyxbxagunkst5d4wx
Feb 17 03:42:21 worker-02 level=info msg="waiting 92.791655ms before registering session" module=node/agent node.id=nie23fzaxyxbxagunkst5d4wx
Feb 17 03:42:22 worker-02 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-03 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-03 level=info msg="forced certificate renewal" module=node/tls node.id=xyjvw1u15ir6mniks70vb4e13 node.role=swarm-worker
Feb 17 03:42:21 worker-03 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.8:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-03 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-03 level=info msg="agent: found node update" module=node/agent node.id=xyjvw1u15ir6mniks70vb4e13
Feb 17 03:42:21 worker-03 level=error msg="agent: session failed" backoff=100ms error="rpc error: code = Canceled desc = context canceled" module=node/agent node.id=xyjvw1u15ir6mniks70vb4e13
Feb 17 03:42:21 worker-03 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-03 level=info msg="manager selected by agent for new session: {9a7k4zuct2uo6ly62dfvdat0d 10.10.10.7:2377}" module=node/agent node.id=xyjvw1u15ir6mniks70vb4e13
Feb 17 03:42:21 worker-03 level=info msg="waiting 78.729389ms before registering session" module=node/agent node.id=xyjvw1u15ir6mniks70vb4e13
Feb 17 03:42:22 worker-03 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-04 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-04 level=info msg="forced certificate renewal" module=node/tls node.id=5rcg19yl5xjstm90fnlsx2m0k node.role=swarm-worker
Feb 17 03:42:21 worker-04 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.3:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-04 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-04 level=info msg="agent: found node update" module=node/agent node.id=5rcg19yl5xjstm90fnlsx2m0k
Feb 17 03:42:21 worker-04 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.8:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-04 level=info msg="manager selected by agent for new session: {7wzpf1et0x8bsuc4788p3loxe 10.10.10.8:2377}" module=node/agent node.id=5rcg19yl5xjstm90fnlsx2m0k
Feb 17 03:42:21 worker-04 level=info msg="waiting 0s before registering session" module=node/agent node.id=5rcg19yl5xjstm90fnlsx2m0k
Feb 17 03:42:22 worker-04 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-05 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-05 level=info msg="forced certificate renewal" module=node/tls node.id=2szicdrvsa227vuwoq82rjnz7 node.role=swarm-worker
Feb 17 03:42:21 worker-05 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-05 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-05 level=info msg="agent: found node update" module=node/agent node.id=2szicdrvsa227vuwoq82rjnz7
Feb 17 03:42:21 worker-05 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.8:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-05 level=info msg="manager selected by agent for new session: {7wzpf1et0x8bsuc4788p3loxe 10.10.10.8:2377}" module=node/agent node.id=2szicdrvsa227vuwoq82rjnz7
Feb 17 03:42:21 worker-05 level=info msg="waiting 0s before registering session" module=node/agent node.id=2szicdrvsa227vuwoq82rjnz7
Feb 17 03:42:22 worker-05 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-07 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-07 level=info msg="forced certificate renewal" module=node/tls node.id=9hom5xlhu5kf6ha2emokt8t2j node.role=swarm-worker
Feb 17 03:42:21 worker-07 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-07 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-07 level=info msg="agent: found node update" module=node/agent node.id=9hom5xlhu5kf6ha2emokt8t2j
Feb 17 03:42:21 worker-07 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.3:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-07 level=info msg="manager selected by agent for new session: {t4tuaimodnterp655mcmvep5s 10.10.10.3:2377}" module=node/agent node.id=9hom5xlhu5kf6ha2emokt8t2j
Feb 17 03:42:21 worker-07 level=info msg="waiting 0s before registering session" module=node/agent node.id=9hom5xlhu5kf6ha2emokt8t2j
Feb 17 03:42:22 worker-07 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-08 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-08 level=info msg="forced certificate renewal" module=node/tls node.id=yrdij8xzlecm3bv68gnfp9enl node.role=swarm-worker
Feb 17 03:42:21 worker-08 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-08 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-08 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.8:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-08 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-08 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-08 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-08 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:22 worker-08 level=info msg="agent: found node update" module=node/agent node.id=yrdij8xzlecm3bv68gnfp9enl
Feb 17 03:42:22 worker-08 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:22 worker-08 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:22 worker-08 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:22 worker-08 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:22 worker-08 level=info msg="manager selected by agent for new session: {9a7k4zuct2uo6ly62dfvdat0d 10.10.10.7:2377}" module=node/agent node.id=yrdij8xzlecm3bv68gnfp9enl
Feb 17 03:42:22 worker-08 level=info msg="waiting 0s before registering session" module=node/agent node.id=yrdij8xzlecm3bv68gnfp9enl
Feb 17 03:42:22 worker-08 level=info msg="initialized VXLAN UDP port to 4789 "

Feb 17 03:42:20 worker-09 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:20 worker-09 level=info msg="forced certificate renewal" module=node/tls node.id=5nee3fbzfvbre11rn74p2nnjs node.role=swarm-worker
Feb 17 03:42:21 worker-09 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.7:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="blockingPicker: the picked transport is not ready, loop back to repick" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-09 level=info msg="initialized VXLAN UDP port to 4789 "
Feb 17 03:42:21 worker-09 level=info msg="agent: found node update" module=node/agent node.id=5nee3fbzfvbre11rn74p2nnjs
Feb 17 03:42:21 worker-09 level=info msg="parsed scheme: \"\"" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="scheme \"\" not registered, fallback to default scheme" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="ccResolverWrapper: sending update to cc: {[{10.10.10.3:2377  <nil> 0 <nil>}] <nil> <nil>}" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="ClientConn switching balancer to \"pick_first\"" module=grpc
Feb 17 03:42:21 worker-09 level=info msg="manager selected by agent for new session: {t4tuaimodnterp655mcmvep5s 10.10.10.3:2377}" module=node/agent node.id=5nee3fbzfvbre11rn74p2nnjs
Feb 17 03:42:21 worker-09 level=info msg="waiting 0s before registering session" module=node/agent node.id=5nee3fbzfvbre11rn74p2nnjs
Feb 17 03:42:22 worker-09 level=info msg="initialized VXLAN UDP port to 4789 "
```

</details>

What do with hung `docker swarm ca --rotate`? Run again? Recreate swarm?

### Reproduce

docker swarm ca --rotate

### Expected behavior

docker swarm ca success rotate

### docker version

```bash
Client: Docker Engine - Community
 Version:           20.10.21
 API version:       1.41
 Go version:        go1.18.7
 Git commit:        baeda1f
 Built:             Tue Oct 25 18:02:28 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.21
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.18.7
  Git commit:       3056208
  Built:            Tue Oct 25 18:00:19 2022
  OS/Arch:          linux/amd64
  Experimental:     true
 containerd:
  Version:          1.6.10
  GitCommit:        770bd0108c32f3fb5c73ae1264f7e503fe7b2661
 runc:
  Version:          1.1.4
  GitCommit:        v1.1.4-0-g5fd4c4d
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)
  scan: Docker Scan (Docker Inc., v0.21.0)

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 0
 Server Version: 20.10.21
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: active
  NodeID: 7wzpf1et0x8bsuc4788p3loxe
  Is Manager: true
  ClusterID: m0ltw0d0vzw528muqqzsavkcc
  Managers: 3
  Nodes: 12
  Default Address Pool: 10.0.0.0/9  172.16.0.0/12
  SubnetSize: 16
  Data Path Port: 4789
  Orchestration:
   Task History Retention Limit: 5
  Raft:
   Snapshot Interval: 10000
   Number of Old Snapshots to Retain: 0
   Heartbeat Tick: 1
   Election Tick: 10
  Dispatcher:
   Heartbeat Period: 5 seconds
  CA Configuration:
   Expiry Duration: 3 months
   Force Rotate: 2
  Autolock Managers: false
  Root Rotation In Progress: true
  Node Address: 10.10.10.8
  Manager Addresses:
   10.10.10.3:2377
   10.10.10.7:2377
   10.10.10.8:2377
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 770bd0108c32f3fb5c73ae1264f7e503fe7b2661
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.0-19-amd64
 Operating System: Debian GNU/Linux 11 (bullseye)
 OSType: linux
 Architecture: x86_64
 CPUs: 1
 Total Memory: 1.937GiB
 Name: manager-02
 ID: JWYA:23DV:TOWZ:RHED:OKBT:UMQM:GXVQ:PMEQ:XTDP:NN2C:T4LW:NALL
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  registry.lan:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by dperny** [comment]
Have you removed a tremendously large number of nodes recently? Like, thousands of nodes removed within the last couple of months?

**Comment by dperny** [comment]
Like, would you have possibly churned 60,000+ nodes in less than a month?

**Comment by dperny** [comment]
I'm strongly suspecting that this is related to #36792.

**Comment by neersighted** [comment]
It appears that the rotate is blocked by down nodes that cannot be removed due to the raft message being too large, and that the rotation is an attempt to resolve that. Normally a rotation blocked by down nodes could be unblocked by simply removing those nodes, but here, things are wedged and so that is not possible.

**Comment by IgorOhrimenko** [comment]
No, I did not remove 60,000+ node last three month.
I fixed prod like that:
```
docker node update --availability pause node-01
docker node rm metal-01
```
but I do not khow how it fixed.

When I pressed CTRL+C after stuck `docker swarm ca --rotate` I got message 'Run swarmctl cluster inspect default to check status' and I tried run the command and get the error: `bash: swarmctl: command not found`. How I can check status?

I checked in the new swarm cluster and get 100% reproduce problem. When swarm has any down nodes it cannot rotate sertificate. 
@neersighted So, 'raft message is too large and can't be sent' does not matter on this issue and rename is not correct, main problem in run the command `docker swarm ca --rotate` with down nodes.

**Comment by neersighted** [comment]
`docker swarm ca --rotate` blocking with down nodes is a feature; you can concurrently remove down nodes with another API client to unblock it if you want to wait for down nodes. Rotating without waiting for down nodes would mean **kicking them out of the cluster**, which is why you must explicitly do so.

The message about `swarmctl` is correct -- it is not part of any distribution of Moby by default and must be built from source, but it will let you directly talk to the Swarm manager to inspect things like pending operations that are not exposed over the REST API.

---

The only sharp edge/gotcha here is the raft message size too large -- we're talking about adding some logging to better confirm the diagnosis when this happens, but after talking through this for an hour, the only situation we can envision where this happens is when the `Cluster` object grows because of a large list of blacklisted certificates, caused by churning nodes.

So, we'd still like to understand why your raft state grew over 1.5Mb; if you haven't churned many (it actually takes around 12k) nodes in the last 30 days, there might still be something else worth investigating.

**Comment by IgorOhrimenko** [comment]
node-01 is freezing about an two hours, LA is too high and network is lag, maybe node-01 sent many messages `i am alive` to manager and than manager lost connect and wrote status down

**Comment by neersighted** [comment]
That's not how the raft state works; the raft state is sent via gRPC to managers only.

The error occurs here:

https://github.com/moby/swarmkit/blob/726f437edbc77e5e283688508281e676b06c48c5/manager/state/raft/raft.go#L1816-L1819

That is where we would want to add some logging.

The InternalRaftRequest in that function is effectively a union:

https://github.com/moby/swarmkit/blob/726f437edbc77e5e283688508281e676b06c48c5/api/raft.proto#L137-L150

Out of these objects, the only one which we expect to grow over time is a `Cluster`; though there are some objects there that are user-controlled like `Secret` -- is it possible you added a Secret over 1.5mb to the cluster?

**Comment by neersighted** [comment]
It looks like `Secret` and `Config` are both enforced to be <= 500kb in the API layer; as such the `Cluster` is the only message here known to have an unbounded size. As there appears to be nothing novel here, I think we can consider this a duplicate of https://github.com/moby/moby/issues/36792.

It would be nice to have better logging/error reporting here to determine definitively which message got too large, but for now I think that's the only actionable aspect of this ticket.

**Comment by neersighted** [comment]
Okay, I don't think there's anything novel here. I'm going to close this out as a duplicate if #36792; adding some logging here to better characterize the failure for future users who stumble on a large `Cluster` object is in the backlog.

**Comment by bluikko** [comment]
> adding some logging here to better characterize the failure for future users who stumble on a large `Cluster` object is in the backlog

Was this ever added?

Cannot remove a node from swarm with the same error message. There are `Config` and `Secret` in the "hundreds" each, but if each of these is limited to 500 kB that should not be over 1.5 MB even if both are at max.  
There seems to be no way to see the various sizes and I am not seeing any extra information in the log.
